### PR TITLE
Add type hints to arelle.XbrlConst

### PR DESCRIPTION
**Pull request by selgamal**
#### Reason for change
Add type hints to arelle/XbrlConst.py

#### Description of change
**This pr produces 260 type checking errors**, this is because currently [`ModelValue.qname`](https://github.com/Arelle/Arelle/blob/8f0b01d86f1cf1933ccc171ce95c93d2488e6ef4/arelle/ModelValue.py#L17) function returns optional QName type to constants that are guaranteed to be QName. With the appropriate overload defined for `qname` function (overload to guarantee returning QName when first param is str), all the errors should disappear.

This pr should be merged after ModelValue type hints are merged.

#### Steps to Test

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Comment by ggravlingen** [comment]
Since this file contains constants, you might want to consider using the [`Final` annotation](https://peps.python.org/pep-0591/#semantics-and-examples).

**Comment by selgamal** [comment]
> Since this file contains constants, you might want to consider using the [`Final` annotation](https://peps.python.org/pep-0591/#semantics-and-examples) where relevant, ie where the constant shouldn't be allowed to be overwritten elsewhere.

Sounds like a good idea, but `Final` was added in python 3.8, and this project still supports python 3.7, is this going to be an issue?

**Comment by ggravlingen** [comment]
It won't be a problem at runtime, I believe, as type hints are not executed at runtime. But you do raise a valid warning so maybe it's a good idea waiting.

**Comment by selgamal** [comment]
> It won't be a problem at runtime, I believe, as type hints are not executed at runtime. But you do raise a valid warning so maybe it's a good idea waiting.

We can add [typing-extensions](https://github.com/python/typing_extensions) to requirements-dev.text, that should resolve this and I think will be useful in general.

File: arelle/XbrlConst.py [hunk 1]
```diff
@@ -1,433 +1,448 @@
...
+    from re import compile as re_compile  # type: ignore[misc]
+
+if TYPE_CHECKING:
+    from lxml import etree
+    from arelle.ModelValue import QName
+    from arelle.ModelObject import ModelObject
+    from arelle.typing import TypeGetText
+    _: TypeGetText
+
+xsd: str = "http://www.w3.org/2001/XMLSchema"
```
[the remark this answers is not in the document]
Comment (selgamal) [reply arelle/XbrlConst.py:20]: @austinmatherne-wk 
No, not needed really, its just a tell that the constant is treated as a `str` in the codebase, if for whatever reason it was changed later, we can definitely live without it.

**Review by sagesmith-wf — APPROVED** [review]

**Review by sagesmith-wf — APPROVED** [review]

File: arelle/ModelValue.py [hunk 2]
```diff
@@ -21,6 +21,18 @@
...
+
+@overload
+def qname(value: ModelObject | str | QName, name: str | ModelObject) -> QName: ...
+
+@overload
+def qname(value: ModelObject | str | QName, name: str | ModelObject | None = None, noPrefixIsNoNamespace: bool = False) -> QName: ...
+
+@overload
+def qname(value: ModelObject | str | QName | Any | None, name: str | ModelObject | None) -> QName | None : ...
+
```
Comment (austinmatherne-wk) [inline arelle/ModelValue.py:35]: Nice!

File: arelle/XbrlConst.py [hunk 3]
```diff
@@ -1,122 +1,151 @@
...
+    from arelle.ModelObject import ModelObject
+    from arelle.typing import TypeGetText
+
+    _: TypeGetText
 
 xsd = "http://www.w3.org/2001/XMLSchema"
-qnXsdSchema = qname("{http://www.w3.org/2001/XMLSchema}xsd:schema")
-qnXsdAppinfo = qname("{http://www.w3.org/2001/XMLSchema}xsd:appinfo")
-qnXsdDefaultType = qname("{http://www.w3.org/2001/XMLSchema}xsd:anyType")
+qnXsdSchema: QName = qname("{http://www.w3.org/2001/XMLSchema}xsd:schema")
```
Comment (austinmatherne-wk) [inline arelle/XbrlConst.py:18]: Thanks to the overloads, can these explicit QName types be removed now?
```suggestion
qnXsdSchema = qname("{http://www.w3.org/2001/XMLSchema}xsd:schema")
```

File: arelle/ModelValue.py (hunk 2 above)
Comment (ggravlingen) [reply arelle/ModelValue.py:35]: Sweet!

**Review by austinmatherne-wk — DISMISSED** [review]

**Comment by austinmatherne-wk** [comment]
@selgamal this PR looks good. If you're happy with it and change it from draft to ready for review I'd be happy to merge it

**Review by austinmatherne-wk — APPROVED** [review]

**Review by sagesmith-wf — APPROVED** [review]

**Pull request merged**